core/src/main/java/kafka/server/share/SharePartition.java

```diff
@@ -2125,16 +2128,25 @@ private long startOffsetDuringInitialization(long partitionDataStartOffset) thro
         if (partitionDataStartOffset != PartitionFactory.UNINITIALIZED_START_OFFSET) {
             return partitionDataStartOffset;
         }
-        GroupConfig.ShareGroupAutoOffsetReset offsetResetStrategy;
+        ShareGroupAutoOffsetResetStrategy offsetResetStrategy;
         if (groupConfigManager.groupConfig(groupId).isPresent()) {
             offsetResetStrategy = groupConfigManager.groupConfig(groupId).get().shareAutoOffsetReset();
         } else {
             offsetResetStrategy = GroupConfig.defaultShareAutoOffsetReset();
         }
 
-        if (offsetResetStrategy == GroupConfig.ShareGroupAutoOffsetReset.EARLIEST)
+        if (offsetResetStrategy.timestamp().isEmpty()) {
+            throw new Exception("The timestamp is not available for the share partition: " + topicIdPartition);
+        }
+        final long timestamp = offsetResetStrategy.timestamp().get();
+
+        if (timestamp == ListOffsetsRequest.LATEST_TIMESTAMP) {
```

Comment:
It's bit odd to fetch the `timestamp` to know which strategy to use however, startegy itself is used to figure out the `timestamp`. Why no to use `offsetResetStrategy` itself?